readme.md

```diff
@@ -33,20 +33,64 @@ const unusedFilename = require('unused-filename');
 
 ## API
 
-### unusedFilename(filePath)
+### unusedFilename(filePath, options?)
 
-Returns a `Promise<string>` containing either the original `filename` or the `filename` appended with a number.
+Returns a `Promise<string>` containing either the original `filename` or the `filename` increment by `options.incrementer`.
 
-### unusedFilename.sync(filePath)
+### unusedFilename.sync(filePath, options?)
 
-Returns a `string` containing either the original `filename` or the `filename` appended with a number.
+Returns a `string` containing either the original `filename` or the `filename` increment by `options.incrementer`.
 
 #### filePath
 
 Type: `string`
 
 The path to check for filename collision.
 
+#### options
+
+Type: `object`
+
+Parameters:
+
+- **incrementer**: `Incrementer` - a function that accepts a file path and increments its index.
+- **maxTries**: `number` - a max number of attempts to take. Default: `Infinity`.
+
+##### incrementer
+
+Type: `(filePath: string) => string`
+Default: Parentheses incrementer (`file.txt` → `file (1).txt`).
+
+A simple function that accepts a file path, and increments its index. It's incrementer's responsibility to extract an already existing index from passed file.
+
+Example incrementer that appends new index as an underscore suffix:
```

Comment:
I would use a different example as this is already possible with our separator incremented. Put the number at the start of the string or something, just to make the example actually useful.